core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerUtils.scala

```diff
@@ -109,20 +109,17 @@ private[mesos] trait MesosSchedulerUtils extends Logging {
 
       new Thread(Utils.getFormattedClassName(this) + "-mesos-driver") {
         setDaemon(true)
+        setUncaughtExceptionHandler(new Thread.UncaughtExceptionHandler {
```

Comment:
I don't think it's clearer in that you'd have to understand what this handler does. A try-catch seems more recognizable